### PR TITLE
data-json - Add conditional template

### PR DESCRIPTION
**Pull request by duboisp**
Add the possibility to do conditional template with the data-json plugin.

Test to do at least:
* [x] Ensure the data-json working example work the same as before
* [x] Ensure the data-json HTML5 template working example work the same as before
* [x] Ensure the JSON manager working example work the same as before
* [x] Test the functional aspect of the conditional template via its working example
* [x] Review the code.

Working example: https://universallabs.org/wet/labs/conditional-tmpl/unmin/demos/wb-data-json/conditional-en.html

### CODE REVIEW

**Comment by duboisp** [comment]
@GormFrank for discussion to know who would be able to review this PR.

**Comment by duboisp** [comment]
Pre-approved upon successful review and local testing by @polmih

File: site/layouts/assessment_wrote_en-en.hbs
Comment (polmih) [inline site/layouts/assessment_wrote_en-en.hbs:186]:
```suggestion
										"test": "fn:guessType",
```

File: site/layouts/assessment_wrote_en-en.hbs
Comment (polmih) [inline site/layouts/assessment_wrote_en-en.hbs:229]:
```suggestion
						"test": "fn:guessType",
```

File: site/layouts/assessment_wrote_en-en.hbs
Comment (polmih) [inline site/layouts/assessment_wrote_en-en.hbs:254]:
```suggestion
									"test": "fn:guessType",
```

File: site/layouts/assessment_wrote_en-en.hbs
Comment (polmih) [inline site/layouts/assessment_wrote_en-en.hbs:264]:
```suggestion
									"test": "fn:guessType",
```

File: site/layouts/assessment_wrote_en-fr.hbs
Comment (polmih) [inline site/layouts/assessment_wrote_en-fr.hbs:185]:
```suggestion
										"test": "fn:guessType",
```

File: site/layouts/assessment_wrote_en-fr.hbs
Comment (polmih) [inline site/layouts/assessment_wrote_en-fr.hbs:194]:
```suggestion
										"test": "fn:guessType",
```

File: site/layouts/assessment_wrote_en-fr.hbs
Comment (polmih) [inline site/layouts/assessment_wrote_en-fr.hbs:207]:
```suggestion
						"test": "fn:guessType",
```

File: site/layouts/assessment_wrote_en-fr.hbs
Comment (polmih) [inline site/layouts/assessment_wrote_en-fr.hbs:219]:
```suggestion
						"test": "fn:guessType",
```

File: site/layouts/assessment_wrote_en-fr.hbs
Comment (polmih) [inline site/layouts/assessment_wrote_en-fr.hbs:228]:
```suggestion
						"test": "fn:guessType",
```

File: site/layouts/assessment_wrote_en-fr.hbs
Comment (polmih) [inline site/layouts/assessment_wrote_en-fr.hbs:253]:
```suggestion
									"test": "fn:guessType",
```

File: site/layouts/assessment_wrote_en-fr.hbs
Comment (polmih) [inline site/layouts/assessment_wrote_en-fr.hbs:263]:
```suggestion
									"test": "fn:guessType",
```

File: src/plugins/wb-data-json/conditional-en.hbs
Comment (polmih) [inline src/plugins/wb-data-json/conditional-en.hbs:32]:
```suggestion
			"test": "fn:guessType",
```

File: src/plugins/wb-data-json/conditional-en.hbs
Comment (polmih) [inline src/plugins/wb-data-json/conditional-en.hbs:309]:
```suggestion
					"template": "[data-guesstype]",
```

File: src/plugins/wb-data-json/conditional-en.hbs
Comment (polmih) [inline src/plugins/wb-data-json/conditional-en.hbs:318]:
```suggestion
					"template": "[data-guesstype]",
```

File: src/plugins/wb-data-json/conditional-en.hbs
Comment (polmih) [inline src/plugins/wb-data-json/conditional-en.hbs:319]:
```suggestion
					"test": "fn:guessType",
```

File: src/plugins/wb-data-json/conditional-en.hbs
Comment (polmih) [inline src/plugins/wb-data-json/conditional-en.hbs:327]:
```suggestion
					"template": "[data-guesstype]",
```

File: src/plugins/wb-data-json/conditional-en.hbs
Comment (polmih) [inline src/plugins/wb-data-json/conditional-en.hbs:328]:
```suggestion
					"test": "fn:guessType",
```

File: src/plugins/wb-data-json/conditional-en.hbs
Comment (polmih) [inline src/plugins/wb-data-json/conditional-en.hbs:336]:
```suggestion
					"template": "[data-guesstype]",
```

Comment (polmih) [reply src/plugins/wb-data-json/conditional-en.hbs:336]: guestType sts all over this file.  I will leave it to you to do a find and replace in the code :)

**Review by polmih — DISMISSED** [review]
All plugins mentioned in the PR description were tested locally. Did not encounter any issues.

**Comment by duboisp** [comment]
Approved once the update is push and a second review / testing is completed.

**Comment by duboisp** [comment]
@GormFrank can you perform the second review? The requested change was applied and fixed the guessType behaviour when its undefined or when its point to a non-typed JSON object.

File: site/pages/docs/ref/wb-data-json/wb-data-json-en.hbs
Comment (GormFrank) [inline site/pages/docs/ref/wb-data-json/wb-data-json-en.hbs:664]: Slight rephrase:
```suggestion
		<td>Template only. Indicate not to initiate the process of iterating the data and instead execute the mapping object directly</td>
```

File: site/pages/docs/ref/wb-data-json/wb-data-json-en.hbs
Comment (GormFrank) [inline site/pages/docs/ref/wb-data-json/wb-data-json-en.hbs:272]: May be worth adding a link to the IRI definition for general audience to get the reference :)

**Review by GormFrank — DISMISSED** [review]
Approved with some content changes that can be tackled after (please consider French pages as well when looking at my comments)

File: site/pages/docs/ref/wb-data-json/wb-data-json-en.hbs
Comment (duboisp) [reply site/pages/docs/ref/wb-data-json/wb-data-json-en.hbs:448]: fixed the French too

File: site/pages/docs/ref/wb-data-json/wb-data-json-en.hbs
Comment (duboisp) [reply site/pages/docs/ref/wb-data-json/wb-data-json-en.hbs:665]: Fixed the French too

File: site/pages/docs/ref/wb-data-json/wb-data-json-en.hbs
Comment (duboisp) [reply site/pages/docs/ref/wb-data-json/wb-data-json-en.hbs:664]: Fixed the French too.

File: site/pages/docs/ref/wb-data-json/wb-data-json-en.hbs
Comment (duboisp) [reply site/pages/docs/ref/wb-data-json/wb-data-json-en.hbs:272]: I added a link to wikipedia about what is an IRI

Comment (duboisp) [reply site/pages/docs/ref/wb-data-json/wb-data-json-en.hbs:272]: And I did the same fix to the French page.

**Review by GormFrank — APPROVED** [review]

**Pull request merged**
